order-service/src/main/java/com/example/orderservice/config/kafka/KafkaStreamsConfig.java

```diff
@@ -58,32 +54,23 @@ class KafkaStreamsConfig {
     }
 
     @Bean
-    StreamsBuilderFactoryBeanConfigurer configurer() {
+    StreamsBuilderFactoryBeanConfigurer configurer(
+            DeadLetterPublishingRecoverer deadLetterPublishingRecoverer) {
         return factoryBean -> {
             factoryBean.setStateListener(
                     (newState, oldState) ->
                             log.info("State transition from {} to {} ", oldState, newState));
+            Properties streamsConfiguration = factoryBean.getStreamsConfiguration();
+            Assert.notNull(streamsConfiguration, "streamsConfiguration must not be null");
+            streamsConfiguration.put(
+                    StreamsConfig.DEFAULT_DESERIALIZATION_EXCEPTION_HANDLER_CLASS_CONFIG,
+                    RecoveringDeserializationExceptionHandler.class);
+            streamsConfiguration.put(
+                    RecoveringDeserializationExceptionHandler.KSTREAM_DESERIALIZATION_RECOVERER,
+                    deadLetterPublishingRecoverer);
         };
     }
 
-    @Bean(KafkaStreamsDefaultConfiguration.DEFAULT_STREAMS_CONFIG_BEAN_NAME)
-    KafkaStreamsConfiguration defaultKafkaStreamsConfig(
-            KafkaConnectionDetails connectionDetails,
-            KafkaProperties kafkaProperties,
-            DeadLetterPublishingRecoverer deadLetterPublishingRecoverer) {
-        Map<String, Object> properties = kafkaProperties.buildStreamsProperties(null);
-        properties.put(
-                ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG,
-                connectionDetails.getStreamsBootstrapServers());
-        properties.put(
-                StreamsConfig.DEFAULT_DESERIALIZATION_EXCEPTION_HANDLER_CLASS_CONFIG,
-                RecoveringDeserializationExceptionHandler.class);
-        properties.put(
-                RecoveringDeserializationExceptionHandler.KSTREAM_DESERIALIZATION_RECOVERER,
-                deadLetterPublishingRecoverer);
-        return new KafkaStreamsConfiguration(properties);
-    }
-
     @Bean
     DeadLetterPublishingRecoverer deadLetterPublishingRecoverer(
```

Comment:
@coderabbitai, please help me to write Integration test to showcase deadLetterPublishingRecover

Reply:
@coderabbitai, received below error but not entering into deadLetterPublishingRecoverer

23:12:53.397 [awaitility-thread] WARN  org.apache.kafka.clients.NetworkClient - [Consumer clientId=consumer-deadletter-test-group-1, groupId=deadletter-test-group] Error while fetching metadata with correlation id 2 : {recovererDLQ=UNKNOWN_TOPIC_OR_PARTITION}
23:12:53.397 [awaitility-thread] INFO  org.apache.kafka.clients.Metadata - [Consumer clientId=consumer-deadletter-test-group-1, groupId=deadletter-test-group] Cluster ID: 4L6g3nShT-eMCtK--X86sw
23:12:53.398 [awaitility-thread] INFO  org.apache.kafka.clients.consumer.internals.ConsumerCoordinator - [Consumer clientId=consumer-deadletter-test-group-1, groupId=deadletter-test-group] Discovered group coordinator localhost:32810 (id: 2147483646 rack: null)
23:12:53.398 [order-service-stream-client-StreamThread-1] ERROR org.apache.kafka.streams.KafkaStreams - stream-client [order-service-stream-client] Encountered the following exception during processing and the registered exception handler opted to SHUTDOWN_CLIENT. The streams client is going to shut down now. 
org.apache.kafka.streams.errors.StreamsException: Exception caught in process. taskId=0_0, processor=KSTREAM-SOURCE-0000000000, topic=payment-orders, partition=0, offset=0, stacktrace=org.apache.kafka.common.errors.SerializationException: Can't serialize data [invalid-payload] for topic [order-service-KSTREAM-JOINTHIS-0000000004-store-changelog]

Reply:
@coderabbitai, can you please help me write code which will handle above possible Next Steps